SpatialGDK/Source/SpatialGDKServices/Private/LocalReceptionistProxyServerManager.cpp

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LocalReceptionistProxyServerManager.h"
+
+#include "Internationalization/Regex.h"
+#include "Sockets.h"
+#include "SocketSubsystem.h"
+#include "SpatialCommandUtils.h"
+#include "UObject/CoreNet.h"
+
+DEFINE_LOG_CATEGORY(LogLocalReceptionistProxyServerManager);
+
+#define LOCTEXT_NAMESPACE "FLocalReceptionistProxyServerManager"
+
+FLocalReceptionistProxyServerManager::FLocalReceptionistProxyServerManager()
+: RunningCloudDeploymentName("")
+{
+}
+
+FString FLocalReceptionistProxyServerManager::GetProcessName()
+{
+	bool bSuccess = false;
+	FString ProcessName = "";
+	const FString TaskListCmd = FString::Printf(TEXT("tasklist"));
+
+	// get the task list line for the process with Pid 
+	const FString TaskListArgs = FString::Printf(TEXT(" /fi \"PID eq %s\" /nh /fo:csv"), *BlockingProcess.Pid);
+	FString TaskListResult;
+	int32 ExitCode;
+	FString StdErr;
+	bSuccess = FPlatformProcess::ExecProcess(*TaskListCmd, *TaskListArgs, &ExitCode, &TaskListResult, &StdErr);
+	if (ExitCode == ExitCodeSuccess && bSuccess)
+	{
+		FRegexPattern ProcessNamePattern(TEXT("\"(.+?)\""));
+		FRegexMatcher ProcessNameMatcher(ProcessNamePattern, TaskListResult);
+		if (ProcessNameMatcher.FindNext())
+		{
+			ProcessName = ProcessNameMatcher.GetCaptureGroup(1 /* Get the Name of the process, which is the first group. */);
+
+			return ProcessName;
+		}
+	}
+
+	UE_LOG(LogLocalReceptionistProxyServerManager, Warning, TEXT("Failed to get the name of the process that is blocking the required port."));
+	
+	return ProcessName;
+}
+
+bool FLocalReceptionistProxyServerManager::CheckIfPortIsBound(int32 Port)
+{
+	const FString NetStatCmd = FString::Printf(TEXT("netstat"));
+
+	// -a display active tcp/udp connections, -o include PID for each connection, -n don't resolve hostnames
+	const FString NetStatArgs = TEXT("-n -o -a");
+	FString NetStatResult;
+	int32 ExitCode;
+	FString StdErr;
+	bool bSuccess = FPlatformProcess::ExecProcess(*NetStatCmd, *NetStatArgs, &ExitCode, &NetStatResult, &StdErr);
+
+	if (ExitCode == ExitCodeSuccess && bSuccess)
+	{
+		// Get the line of the netstat output that contains the port we're looking for.
+		FRegexPattern PidMatcherPattern(FString::Printf(TEXT("(.*?:%i.)(.*)( [0-9]+)"), Port));
+		FRegexMatcher PidMatcher(PidMatcherPattern, NetStatResult);
+		if (PidMatcher.FindNext())
+		{
+			const FString State = PidMatcher.GetCaptureGroup(2 /* Get the State of the process, which is the second group. */);
+			if (State.Contains("LISTENING"))
+			{
+				
+				BlockingProcess.Pid = PidMatcher.GetCaptureGroup(3 /* Get the PID, which is the third group. */);
+				BlockingProcess.Name = GetProcessName();
```

Comment:
Wonder if we should check that GetProcessName() has succeeded in getting ProcessName?

Reply:
So because it will return an empty string in case it can not find the name and because I am using it only for logging purposes I thought it will be okay as at the Log we can just skip the name of process. Example: 
`"Process Spatial.exe with PID:%s is blocking required port."` or `"Process  with PID:%s is blocking required port."`  But maybe that is not the best approach.

Reply:
changed it to check if we get the process name